# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [4]:
library(tidyverse)

# Assignment 1

First, we generate some random data:

In [5]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6        V7      
1 0.7587035 0.5059483 0.4544810 0.2613519 0.7957272 0.5391732 0.387087
2 0.7260426 0.7669636 0.9426186 0.2713762 0.5776037 0.3388035 0.878919
  V8        V9        V10       V11        V12       V13        V14      
1 0.6159734 0.3740715 0.8806425 0.03966291 0.4221153 0.51474216 0.9125247
2 0.1687844 0.3292634 0.4103291 0.21049382 0.7657696 0.02303574 0.1533197
  V15       V16      V17       V18        V19       V20      
1 0.4630755 0.124926 0.2362454 0.07897074 0.2012728 0.8503375
2 0.4693674 0.643678 0.2308258 0.18744445 0.4329738 0.8319108

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [6]:
df2 <- df1 %>%
  gather(Key, Value, V1:V20)

head(df2)

Key Value    
1 V1  0.7587035
2 V1  0.7260426
3 V2  0.5059483
4 V2  0.7669636
5 V3  0.4544810
6 V3  0.9426186

## Assignment 2

We will generate some data:

In [7]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [8]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [9]:
df5 <- full_join(df3, df4, by = 'id')
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook